src/modules/profile/profile.controller.ts

```diff
@@ -17,11 +18,63 @@ export class ProfileController {
     return this.profileService.getAllSkills();
   }
 
+  @Get('allItem')
+  getAllProfile() {
+    return this.profileService.getAllProfile();
+  }
+
+  @Get('filter')
+  async findAll(@Query() userQuery: FindUserDto) {
+    const profileInfo = await this.profileService.queryBuilderSkills('skillsprofile');
+    const filterProfile = await this.profileService.paginationFilter(userQuery, profileInfo);
+    const user = await this.profileService.queryBuilderUser('userprofile');
```

Comment:
replase with one query. its not optimal way of quering

Reply:
fixed

Reply:
fixed